crates/bitwarden-exporters/src/cxp/export.rs

```diff
@@ -0,0 +1,370 @@
+            CipherType::Identity(_) => ItemType::Identity,
+            CipherType::SecureNote(_) => ItemType::Document,
+            CipherType::SshKey(_) => todo!(),
+        }
+    }
+}
+
+impl From<CipherType> for Vec<Credential> {
+    fn from(value: CipherType) -> Self {
+        match value {
+            CipherType::Login(login) => (*login).into(),
+            CipherType::Card(_) => vec![],
+            CipherType::Identity(_) => vec![],
+            CipherType::SecureNote(_) => vec![],
+            CipherType::SshKey(_) => vec![],
+        }
+    }
+}
+
+impl From<Login> for Vec<Credential> {
+    fn from(login: Login) -> Self {
+        let mut credentials = vec![];
+
+        if login.username.is_some() || login.password.is_some() || !login.login_uris.is_empty() {
+            credentials.push(Credential::BasicAuth(login.clone().into()));
+        }
+
+        if let Some(totp) = login.totp {
+            if let Ok(totp) = totp.parse::<Totp>() {
+                // TODO(PM-15389): Properly set username/issuer.
+                credentials.push(Credential::Totp {
+                    secret: totp.secret.into(),
+                    period: totp.period as u8,
+                    digits: totp.digits as u8,
+                    username: "".to_string(),
+                    algorithm: match totp.algorithm {
+                        TotpAlgorithm::Sha1 => OTPHashAlgorithm::Sha1,
+                        TotpAlgorithm::Sha256 => OTPHashAlgorithm::Sha256,
+                        TotpAlgorithm::Sha512 => OTPHashAlgorithm::Sha512,
+                        TotpAlgorithm::Steam => OTPHashAlgorithm::Unknown("steam".to_string()),
+                    },
+                    issuer: None,
+                })
+            }
+        }
+
+        if let Some(fido2_credentials) = login.fido2_credentials {
+            for fido2_credential in fido2_credentials {
+                let c = fido2_credential.try_into();
+                if let Ok(c) = c {
+                    credentials.push(Credential::Passkey(c))
+                }
+            }
+        }
+
+        credentials
+    }
+}
+
+impl From<Login> for BasicAuthCredential {
+    fn from(login: Login) -> Self {
+        BasicAuthCredential {
+            urls: login
+                .login_uris
+                .into_iter()
+                .flat_map(|uri| uri.uri)
+                .collect(),
+            username: login.username.map(|value| EditableField {
+                id: random_id(),
+                field_type: FieldType::String,
+                value,
+                label: None,
+            }),
+            password: login.password.map(|value| EditableField {
+                id: random_id(),
+                field_type: FieldType::ConcealedString,
+                value,
+                label: None,
+            }),
+        }
+    }
+}
+
+/// Generate a 32 byte random ID
+fn random_id() -> B64Url {
+    generate_random_bytes::<[u8; 32]>().as_slice().into()
+}
```

Comment:
Todo: Add a comment that we should be able to remove this in the near future as the spec will most likely change the IDs to be optional.

Need to figure out when Apple will change their API though.